src/vm/securitydescriptor.cpp

```diff
@@ -476,4 +476,20 @@ BOOL PEFileSecurityDescriptor::AllowBindingRedirects()
 
 #endif // FEATURE_CORECLR
 
+#if defined(FEATURE_COLLECTIBLE_ALC) && !defined(CROSSGEN_COMPILE)
+SecurityDescriptor::~SecurityDescriptor()
+{
+    // When the AppDomain's LoaderAllocator is not collectible, its LargeHeapHandleTable will
+    // be used for the permission handles
+    if (!m_pLoaderAllocator->IsCollectible())
+    {
+        if (m_hGrantedPermissionSet != NULL)
+            m_pAppDomain->ReleaseObjRefPtrsInLargeTable((OBJECTREF*)((UINT_PTR)m_hGrantedPermissionSet - 1), 1);
```

Comment:
This should not be needed. The handle in `StoreObjectInLazyHandle` should be allocated from collectible `LoaderAllocator` and automatically reclaimed once the collectible `LoaderAllocator` is destroyed.

Otherwise, you will have this memory leak problem in every place where `LoaderAllocator::AllocateHandle` is called.